bpf/lib/egress_policies.h

```diff
@@ -579,6 +580,59 @@ srv6_store_meta_sid(struct __ctx_buff *ctx, const union v6addr *sid)
 	ctx_store_meta(ctx, CB_SRV6_SID_4, sid->p4);
 }
 
+#ifdef ENABLE_IPV6
+/* SRv6 encapsulation occurs at the native-dev currently.
+ * Its possible that after encapsulation a fib entry exists which would actually
+ * route the IPv6 destination somewhere else.
+ *
+ * Therefore, this function performs an additional fib lookup on the encap'd
+ * packet to ensure we transmit it via the correct link and with the correct
+ * l2 addresses.
+ */
+static __always_inline int
+srv6_refib(struct __ctx_buff *ctx)
+{
+	struct bpf_fib_lookup_padded params = {0};
+	__u32 old_oif = ctx_get_ifindex(ctx);
+	void *data, *data_end;
+	struct ipv6hdr *ip6;
+	__s8 fib_ret;
+
+	if (!revalidate_data(ctx, &data, &data_end, &ip6))
+		return DROP_INVALID;
+
+	fib_ret = (__s8)fib_lookup_v6(ctx, &params, ip6, BPF_FIB_LOOKUP_OUTPUT);
+
+	if (fib_ret && fib_ret != BPF_FIB_LKUP_RET_NO_NEIGH)
+		return DROP_NO_FIB;
+
```

Comment:
You could just fold this into the `switch()`  statement as the `default` case :).

Reply:
Yeah I like that, will change.